assets/js/components/Dialog.js

```diff
@@ -109,9 +111,14 @@ const Dialog = ( {
 								<Button
 									onClick={ handleConfirm }
 									danger={ danger }
+									disabled={ showSpinner }
 								>
 									{ confirmButton ? confirmButton : __( 'Disconnect', 'google-site-kit' ) }
 								</Button>
+								<Spinner
+									className="googlesitekit-margin-right-auto"
+									isSaving={ showSpinner }
+								/>
 								<Link
 									className="mdc-dialog__cancel-button"
 									onClick={ handleDialog }
```

Comment:
Let's also disable the `Link` here using the same logic as the button – our link component supports the `disabled` prop even though normal `<a>` tags don't 😉